### ISSUE TITLE
Let user specify network name for autoscaling in Docker Swarm

### ISSUE THREAD

**Issue by vicary** [emqtt/emq_docker#84]
Make sure you read and understand http://emqtt.io/docs/v2/index.html.
Use one of the two templates below and delete the rest.

- [ ] BUG REPORT
- [x] FEATURE REQUEST

-------------------------------------------------------------------------------
> FEATURE REQUEST

### Missing feature
Able to specify a network interface in env var, then `start.sh` or `cluster.sh` uses the specified network to resolve the default IP into `$LOCAL_IP`.

devOps should be able to dynamically allocate `EMQ_HOST` or `EMQ_NODE__NAME` in a deterministic way, since Docker services with multiple networks will attach interfaces in random order.

### Justification
With multiple networks attached, the current way to get `LOCAL_IP` in `start.sh` is not deterministic; While IP is generated *after* the container is spawn.

### Workarounds
Are there any workarounds you currently have in place because the feature is missing?

In short, no.

The overlay network address is load balanced and cannot resolve single nodes, individual nodes spawned need a way to resolve their own unique address correctly, especially when there are more than one interface and/or IP available.

**Comment by ozbillwang** [comment]
@vicary 

Did you get any solution on it? 

The exist solution to join cluster, doesn't really work. 

For example, I have aws load balancer, 

```
DNS (q.mqtt.com) -> aws elb (8883) -> ssl offload |-> node1 (1883)
                                                  |-> node2 (1883)
```
What cluster should I join? DNS name, elb end point, or others?

Second, the note for cluster

>Note: You must publsh port 4369 and range of port 6000-6999 for EMQ Clustered.

So does it mean I have to open multiple port , 4369 and 6000-6999 on elb ? Currently aws elb doesn't support to listen multiple ports. I have to set the port redirection one by one

**Comment by vicary** [comment]
@ozbillwang Not sure if you're spawning them in docker swarm. If so, ELB may not be able to route inside the swarm ingress network. You could either use host networking for your containers, or spin up something like Traefik to route external connections from ELB to container IP's or service networks.
